react/src/layout/multipanel.tsx

```diff
@@ -0,0 +1,75 @@
+import React from "react";
+import { CModelsWrapper } from "../data/wrapper";
+import { Col, Row } from "react-bootstrap";
+import { LAYOUTS } from "./layouts";
+import { LayoutProps, Layout } from "./layout";
+import { Panel } from "../component/reusable/panel";
+import { ReportAnimationController, useAnimationReader } from "./report"
+import { SchemaLayout } from "../utility/schema";
+import { useContext } from "react";
+import { useInterpolatedString, ValueSelectors } from "../hook/string";
+import { useWindowSize } from "../hook/breakpoint";
+
+export type MultiPanelConfig = {
+    items: SchemaLayout[],
+    title: string,
+    reportName: string,
+    reportGroupName: string,
+    frameIdFields: string[],
+}
+
+export class MultiPanelLayout extends Layout<MultiPanelConfig, {}> {
+    items: Layout[];
+
+    constructor(config: MultiPanelConfig) {
+        super(config);
+        this.items = config.items.map(LAYOUTS.getLayoutForSchema);
+    }
+
+    getFormDependencies = () => {
+        return [...(this.items)].map(childLayout => childLayout.getFormDependencies()).flat();
+    }
+
+    component = (props: LayoutProps<{}>) => {
+        let { reportName, reportGroupName, frameIdFields } = this.config;
+        let modelsWrapper = useContext(CModelsWrapper);
+        let title = useInterpolatedString(modelsWrapper, this.config.title, ValueSelectors.Models);
+        //TODO(pjm): without this call, heatplots don't respond to window resize
```

Comment:
This is fine, I think we should think about hooks as having side-effects. This is very declarative and works.